### DISCUSSION TITLE
Steps to run Adapter and Processing-ms microservice

### DISCUSSION THREAD

**Discussion by GitVivekHub** [Sunbird-cQube/community#789] [Q&A]
1.  I have uploaded the national programs zip file via ingestion/national_programs api. I can see my  zip file inside the emission folder on minio bucket. Now i want to run the adapter for it so that i can generate events and dimension  csv based on metrics.
What steps do i have to consider for this to run the adapter.

2. After running the adapter if i want to insert the data from the csv on the bucket into the database what steps inside processing-ms do i have to consider .

As per the documentation there are yarn cli ingest commands.

Now do i have to run these commands explicitly or if i just schedule the adapter via spec/schedule api the adapter will run the NIFI process automatically ?

**Comment by pandutibil** [comment]
@GitVivekHub 

First, run the Adapters NIFI processor group using schedule API, then "process_input" folder is created with the program names in the minio bucket.
Here's the **processor_group_name**: "Run_adapters"

After successfully files are added to the minio bucket for processing, then run another NIFI processor group to process and stores the data in the database using Schedule API.
Here's the **processor_group_name**: "Run the Latest Code local"
Note: This processor group will automatically run the ingestion commands.
